src/controllers/order.conrtoller.ts

```diff
@@ -0,0 +1,97 @@
+import * as Koa from 'koa'
+import * as HttpStatus from 'http-status-codes'
+import { getRepository, Repository } from 'typeorm'
+
+import Order from '../models/order.entity'
+import Dish from '../models/menu.entity'
+import CartPosition from '../models/cart.entity'
+
+export default class OrderController {
+  static async addOrder(ctx: Koa.Context) {
+    const cartRepo: Repository<CartPosition> = getRepository(CartPosition)
+    const orderRepo: Repository<Order> = getRepository(Order)
+
+    const cart: CartPosition[] = await cartRepo.find({
+      where: {
+        user: ctx.params.user_id,
+      }, 
+      relations: ['dish']
+    })
+
+    const dishes: Dish[] = cart.map((current) => {
+      return current.dish
+    })
+
+    const order: Order = orderRepo.create({
+      dishes: dishes,
+      state: ctx.request.body.state,
```

Comment:
looks a little overwhelmed
I mean let's use destructuring here or `...ctx.request.body`

Reply:
cannot adjust as proposed due to typization issues